project/frontend/src/services/auth.js

```diff
@@ -0,0 +1,81 @@
+import firebase from "firebase/app";
+import "firebase/auth";
+import firebaseConfig from "./config/firebaseConfig";
+import "regenerator-runtime/runtime";
+
+firebase.initializeApp(firebaseConfig);
+firebase.auth().setPersistence(firebase.auth.Auth.Persistence.NONE);
+
+const postIdTokenToAuth = (idToken) => {
+  return fetch("/authentication", {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: idToken,
+  });
+};
+
+const googleProvider = new firebase.auth.GoogleAuthProvider();
+const signIn = (callback) => {
+  firebase
+    .auth()
+    .signInWithPopup(googleProvider)
+    .then((result) => {
+      let user = result.user;
+      user.getIdToken().then((idToken) => {
+        postIdTokenToAuth(idToken)
+          .then((response) => response.json())
+          .then((data) => {
+            callback(true, {
+              email: user.email,
+              displayName: user.displayName,
+              id: data.id,
+            });
+            console.log("user id: " + data.id);
+          });
+      });
+    });
```

Comment:
we should add a `catch` here to feed a `false` to the `callback` function if anything in the flow fails @Phoebe-Khokgawe

Reply:
yup

Reply:
Added here 09661af7fbdaf46c72494d7deecf4fcdba1f89c1
Will it be better to add it to all the promises or just the final one here is enough.